tests/release_tests/test_launch/test_kubernetes.py

```diff
@@ -0,0 +1,160 @@
+import os
+
+import pytest
+import yaml
+from utils import (
+    cleanup_deployment,
+    create_wandb_and_aws_secrets,
+    get_wandb_api_key,
+    run_cmd,
+    run_cmd_async,
+    setup_cleanup_on_exit,
+    wait_for_image_job_completion,
+    update_dict,
+)
+from wandb.sdk.launch.launch_add import launch_add
+
+NAMESPACE = "wandb-release-testing"
+ENTITY = "launch-release-testing"
+PROJECT = "release-testing"
+QUEUE = "kubernetes-queue"
+JOB_NAME = "sample_job:v0"  # simple job that counts to 50
+
+LAUNCH_JOB_CONFIG = {
+    "resource_args": {"kubernetes": {"namespace": NAMESPACE}},
+}
+
+AWS_CREDENTIALS_FORMAT = """[default]
+aws_access_key_id = {aws_id}
+aws_secret_access_key = {aws_secret}
+aws_session_token = {aws_token}
+"""
+
+
+@pytest.mark.timeout(180)
+def test_kubernetes_agent_on_local_process(api_key, base_url):
+    if not api_key:
+        api_key = get_wandb_api_key(base_url)
+    aws_id = os.getenv("AWS_ACCESS_KEY_ID")
+    aws_secret = os.getenv("AWS_SECRET_ACCESS_KEY")
+    aws_token = os.getenv("AWS_SESSION_TOKEN")
+
+    with open(os.path.expanduser("~/.aws/credentials"), "w") as f:
+        f.write(
+            AWS_CREDENTIALS_FORMAT.format(
+                aws_id=aws_id, aws_secret=aws_secret, aws_token=aws_token
+            )
```

Comment:
I think this (plus the other one below) isn't an issue since it takes these credentials straight from the runner's machine and writes them back into their own credentials file.

Reply:
Actually I think doing a `aws sts` call from the CLI will make this unnecessary, I'll look into it more and update the readme